File: Python/Task12/n23912.py
﻿# https://inf-ege.sdamgia.ru/problem?id=23912

s = '>' + '1'*10 + '2'*20 + '3'*30
while '>1' in s or '>2' in s or '>3' in s:
	if '>1' in s:
		s = s.replace('>1', '22>', 1)
	elif '>2' in s:
		s = s.replace('>2', '2>', 1)
	elif '>3' in s:
		s = s.replace('>3', '1>', 1)
print(1*s.count('1') + 2*s.count('2') + 3*s.count('3'))

File: C++/Task12/n23912/n23912.cpp
// https://inf-ege.sdamgia.ru/problem?id=23912

#include <iostream>

using namespace std;

int main()
{
	string s = string(1, '>') + string(10, '1') + string(20, '2') + string(30, '3');
	while (s.find(">1") != string::npos
		|| s.find(">2") != string::npos
		|| s.find(">3") != string::npos)
	{
		size_t pos = s.find(">1");
		if (pos != string::npos)
		{
			s.replace(pos, 2, "22>");
		}
		pos = s.find(">2");
		if (pos != string::npos)
		{
			s.replace(pos, 2, "2>");
		}
		pos = s.find(">3");
		if (pos != string::npos)
		{
			s.replace(pos, 2, "1>");
		}
	}
	long long c = 
		count(s.begin(), s.end(), '1') +
		count(s.begin(), s.end(), '2') * 2 +
		count(s.begin(), s.end(), '3') * 3;
	std::cout << c << std::endl;
	return 0;
}